chasm/visibility.go

```diff
@@ -31,6 +32,41 @@ type VisibilityMemoProvider interface {
 	Memo(Context) map[string]VisibilityValue
 }
 
+type VisibilitySearchAttributesMapper struct {
```

Comment:
Does this type need to be exported? If so, please document it and all of its methods.